modules/objdetect/src/aruco/fractal_detector.cpp

```diff
@@ -0,0 +1,954 @@
+
+                    Kdtree.radiusSearch(query, indices, dists, 400.0, 1, cv::flann::SearchParams());
+
+                    if (indices.empty()) continue;
+                    int nearestIdx = indices[0];
+
+                    float newDist = cv::norm(cv::Point2f(kpoints[nearestIdx].pt) - cv::Point2f(imgPoints[idx]));
+
+                    if (kpoints[nearestIdx].class_id != objKeyPoints[idx].class_id || dists[0] > 320 || dists[0] == 0) {
+                        continue;
+                    }
+                    if (nearestIdx != -1) {
+                        bool duplicateFound = false;
+                        for (size_t i = 0; i < nearestIdxList.size(); ++i) {
+                            if (nearestIdxList[i] == nearestIdx) {
+                                duplicateFound = true;
+                                float existingDist = distsList[i];
+                                if (newDist < existingDist) {
+                                    p2d_vec[i] = kpoints[nearestIdx].pt;
+                                    p3d_vec[i] = cv::Point3f(objPoints[idx].x, objPoints[idx].y, 0);
+                                    distsList[i] = newDist; // update distsList
+                                }
+                                break;
+                            }
+                        }
+
+                        if (!duplicateFound) {
+                            nearestIdxList.push_back(nearestIdx);
+                            distsList.push_back(newDist);
+
+                            p2d_vec.push_back(kpoints[nearestIdx].pt);
+                            p3d_vec.push_back(cv::Point3f(objPoints[idx].x, objPoints[idx].y, 0));
+                        }
+                    }
+                }
+            }
+        }
+        else {
+            // If a marker is detected and it is not possible to take all their corners,
+            // at least take the external one!
+            for (auto markerDetected : markers) {
+                if (markerDetected.fractalMarkerImpl->id == fm.first) {
+                    for (int c = 0; c < 4; c++) {
+                        cv::Point2f pt = markerDetected.fractalMarkerImpl->keypts[c].pt;
+                        p3d_vec.push_back(cv::Point3f(pt.x, pt.y, 0));
+                        p2d_vec.push_back(markerDetected[c]);
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    // Subpixel refinement
+    if (!p2d_vec.empty()) {
+        
+        cv::Size winSize(4, 4);
+        cv::Size zeroZone(-1, -1);
+        cv::TermCriteria criteria(cv::TermCriteria::MAX_ITER | cv::TermCriteria::EPS, 12, 0.005);
+        cornerSubPix(bwimage, p2d_vec, winSize, zeroZone, criteria);
+        cv::Mat(p2d_vec).copyTo(p2d);
+        cv::Mat(p3d_vec).copyTo(p3d);
+    }
+    return true;
+}
+
+std::vector<cv::Point2f> FractalDetector::FractalDetectorImpl::sort(const std::vector<cv::Point2f>& marker) {
```

Comment:
I propose to rename the method as it's not sorting actually.  The idea is to reorder corners to have points clockwise or conter-clockwise. It should be in function name IMHO.